src/main/java/spoon/support/sniper/internal/AbstractSourceFragmentPrinter.java

```diff
@@ -102,6 +102,11 @@ public int update(PrinterEvent event) {
 			//so skip printing of this comment
 			//comment will be printed at place where it belongs to - together with spaces
 			return -1;
+		} else if (event.getRole() == CtRole.DECLARED_IMPORT && fragmentIndex == 0) {
+			// this is the first pre-existing import statement, and so we must print all newline
+			// events unconditionally to avoid placing it on the same line as a newly added import
+			// statement. See PR #3702 for details
+			printStandardSpaces();
```

Comment:
This is the special handling of the first import statement. We may need to add more roles here as more problems crop up. Using only the condition `fragmentIndex == 0` causes unwanted newlines in if/else statements, and perhaps elsewhere as well.